api/src/main/java/org/pmiops/workbench/api/SumoLogicController.java

```diff
@@ -0,0 +1,102 @@
+package org.pmiops.workbench.api;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Logger;
+import org.pmiops.workbench.actionaudit.auditors.SumoLogicAuditor;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.pmiops.workbench.exceptions.UnauthorizedException;
+import org.pmiops.workbench.google.CloudStorageService;
+import org.pmiops.workbench.model.EgressEvent;
+import org.pmiops.workbench.model.EgressEventRequest;
+import org.pmiops.workbench.model.EmptyResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class SumoLogicController implements SumoLogicApiDelegate {
+
+  public static final String SUMOLOGIC_KEY_FILENAME = "inbound-sumologic-keys.txt";
+
+  private static final Logger log = Logger.getLogger(SumoLogicController.class.getName());
+  private final SumoLogicAuditor sumoLogicAuditor;
+  private final CloudStorageService cloudStorageService;
+
+  @Autowired
+  SumoLogicController(SumoLogicAuditor sumoLogicAuditor, CloudStorageService cloudStorageService) {
+    this.sumoLogicAuditor = sumoLogicAuditor;
+    this.cloudStorageService = cloudStorageService;
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> logEgressEvent(
+      String X_API_KEY, EgressEventRequest request) {
```

Comment:
why are we screaming the parameter name?

Reply:
Swagger-codegen is choosing the parameter naming here – it's based on what I used in the YAML file. It felt more appropriate to use the traditional UPPERCASE_HEADER casing over there, so... meh.